### PR TITLE
feat: add prefix to UiListItem

### PR DESCRIPTION
**Pull request by pspaczek**
## Description
This PR:
* add the `UiListItemAffix` component as the base for `UiListItemPrefix` and `UiListItemSuffix`
* use `UiListItemAffix` to create `UiListItemPrefix`
* refactor `UiListItemSuffix`. Use `UiListItemAffix` and handle `@deprecated` props `icon` to keep transparent API for prefix and suffix component
* handle hover and active states of affix
* update stories, and adjust them to UiListItem design

## Related Issue

Closes #

## Motivation and Context

One design of Default UiMenuItem has a prefix element. Same as the design for  UiListItem "32 Icon List Item". To allow the use both designs I add the prefix to UiListItem
**UiListItem**
Zrzut ekranu 2024-04-22 o 12 36 44
UiMenuItem
Zrzut ekranu 2024-04-22 o 12 37 21

## How Has This Been Tested?

Null
## Screenshots (if appropriate):

Zrzut ekranu 2024-04-25 o 10 36 09
Zrzut ekranu 2024-04-25 o 10 36 20
Zrzut ekranu 2024-04-25 o 10 36 36
Zrzut ekranu 2024-04-25 o 10 39 08

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/components/organisms/UiList/_internal/UiListItemSuffix.vue [hunk 1]
```diff
@@ -60,7 +60,7 @@ export interface ListItemSuffixProps {
   /**
    * Use this props to set list item suffix tag.
    */
-  tag?: HTMLTag;
+  tag?: string | HTMLTag;
```
Comment (AlanLes) [inline src/components/organisms/UiList/_internal/UiListItemSuffix.vue:63]: Isn't HTMLTag a narrowed string type already? Why this change? This type doesn't make sense because the `string` type already includes `HTMLTag`. I see the reason - because of the default `''` value - but IMO, it's not a correct solution. Its default value should be undefined (or null; it depends), but not `''`.

File: src/components/organisms/UiList/_internal/UiListItemPrefix.vue [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+  HTMLTag,
+  Icon,
+} from '@/types';
+import UiIcon, { type IconAttrsProps } from '../../../atoms/UiIcon/UiIcon.vue';
+
+export interface ListItemPrefixProps {
+  /**
+   * Use this props to set list item prefix tag.
+   */
+  tag?: string | HTMLTag;
```
Comment (AlanLes) [inline src/components/organisms/UiList/_internal/UiListItemPrefix.vue:60]: And the same here.

**Review by AlanLes — COMMENTED** [review]
Overall looks good 👍 
Have you checked it locally with the MGP? Does it break the app or doesn't it?

**Comment by pspaczek** [comment]
> Overall looks good 👍 Have you checked it locally with the MGP? Does it break the app or doesn't it?

tested in MGP, it doesn't break app

File: src/components/organisms/UiList/_internal/UiListItem.vue [hunk 3]
```diff
@@ -110,6 +140,9 @@ const filteredAttrs = computed(() => {
 const suffixComponent = computed(() => (props.hasSuffix
   ? defineAsyncComponent(() => import('./UiListItemSuffix.vue'))
   : null));
+const prefixComponent = computed(() => (props.hasPrefix
```
Comment (kubajmarek) [inline src/components/organisms/UiList/_internal/UiListItem.vue:143]: Same reorder comment here.

File: src/components/organisms/UiList/_internal/UiListItemPrefix.vue
Comment (kubajmarek) [inline src/components/organisms/UiList/_internal/UiListItemPrefix.vue:1]: Taking a quick look at `ItemPrefix` and `ItemSuffix`, aren't they pretty much identical? The only difference I could find is the reorder of icon & label. 🤔

**Review by AlanLes — COMMENTED** [review]
It could be simplified even more. The `Prefix` and `Suffix` are still almost identical, with slight differences in the template. We could get rid of `UiListItemPrefix` and `UiListItemSuffix` and use the `UiListItemAffix` straight away in the UiList, couldn't we? The only difference between those two is the elements' order: `icon, label` vs. `label, icon`

**Comment by pspaczek** [comment]
> It could be simplified even more. The `Prefix` and `Suffix` are still almost identical, with slight differences in the template. We could get rid of `UiListItemPrefix` and `UiListItemSuffix` and use the `UiListItemAffix` straight away in the UiList, couldn't we? The only difference between those two is the elements' order: `icon, label` vs. `label, icon`

Order and CSS Variables for both i.e --list-item-item-suffix-icon-color. We can't merge it to --ui-list-item-affix

**Comment by pspaczek** [comment]
@AlanLes now duplicate parts are in the affix component. For Prefix just add template and CSS Variables for suffixes and prefixes. The suffix has @deprecated props. This way allows us to make changes in the template in the feature.

**Review by sonya0504 — APPROVED** [review]

File: src/components/organisms/UiMultipleChoices/_internal/UiMultipleChoicesItem.vue
Comment (pspaczek) [inline src/components/organisms/UiMultipleChoices/_internal/UiMultipleChoicesItem.vue:1]: adjustment to changes in UiListItem styles

**Review by sonya0504 — APPROVED** [review]

**Review by AlanLes — APPROVED** [review]
OK, nothing more to add ;)
